prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/Prometheus.scala

```diff
@@ -191,6 +194,45 @@ object Prometheus {
   private val DefaultHistogramBuckets: NonEmptyList[Double] =
     NonEmptyList(.005, List(.01, .025, .05, .075, .1, .25, .5, .75, 1, 2.5, 5, 7.5, 10))
 
+  /**
+    * Given an exclude function, return a 'classifier' function, i.e. for application in
+    * [[org.http4s.server/client.middleware.Metrics#apply]].
+    *
+    * Let's say you want a classifier that excludes integers since your paths consist of:
+    *   * GET    /users/{integer} = get_users_*
+    *   * POST   /users           = post_users
+    *   * PUT    /users/{integer} = put_users_*
+    *   * DELETE /users/{integer} = delete_users_*
+    *
+    * In such a case, we could use:
+    *
+    * classifierFMethodWithOptionallyExcludedPath { str: String => scala.util.Try(str.toInt).isSuccess }
+    *
+    * @param exclude For a given String, namely a path value, determine whether the value gets excluded.
+    * @return Request[F] => Option[String]
+    */
+  def classifierFMethodWithOptionallyExcludedPath[F[_]](
+      exclude: String => Boolean
+  ): Request[F] => Option[String] = { request: Request[F] =>
+    val dsl: Http4sDsl[F] = Http4sDsl[F]
```

Comment:
I'm surprised http4s-dsl is available in this project.  That seems like a bad coupling.  Can we do this without it?

Reply:
> I'm surprised http4s-dsl is available in this project. That seems like a bad coupling.

I only used it since I saw it available via https://github.com/http4s/http4s/blob/series/0.20/build.sbt#L153. 

> Can we do this without it?

Sure. Can you please advise if you know, off-hand, how to get the path as a `List[String]`, @rossabaker?

Reply:
You could copy [this code](https://github.com/http4s/http4s/blob/0801749674a83afa3e7522f4e402d1bedea76982/dsl/src/main/scala/org/http4s/dsl/impl/Path.scala#L57-L64) and build a `List` rather than a `Path`.

In the long run, turning a path into a list of segments ought to be core functionality on a URI.

Reply:
So this dependency already exists, to support the `PrometheusExportService`.  I think we could re-express it without that, but that's a separate ticket.

Reply:
Thanks, Ross, for telling me how to get a path list from a `String`. I did that, as well as removed the `Http4sDsl` usage via commits at https://github.com/http4s/http4s/pull/3167#commits-pushed-1c3a145.